src/components/collapsible_nav/collapsible_nav.tsx

```diff
@@ -65,40 +85,65 @@ export const EuiCollapsibleNav: FunctionComponent<EuiCollapsibleNavProps> = ({
   };
 
   const collapse = () => {
-    if (!isDocked) {
-      onClose();
+    if (!navIsDocked) {
+      onClose && onClose();
     }
   };
 
   const classes = classNames(
     'euiCollapsibleNav',
-    { 'euiCollapsibleNav--isDocked': isDocked },
+    { 'euiCollapsibleNav--isDocked': navIsDocked },
     className
   );
 
   let optionalOverlay;
-  if (!isDocked) {
+  if (!navIsDocked) {
     optionalOverlay = <EuiOverlayMask onClick={collapse} />;
   }
 
-  return (
-    <Fragment>
+  // Show a trigger button if one was passed but
+  // not if hideButtonIfDocked and navIsDocked
+  const trigger =
+    button &&
+    !(hideButtonIfDocked && navIsDocked) &&
+    cloneElement(button as ReactElement, {
+      'aria-controls': flyoutID,
+      'aria-expanded': isOpen,
+      'aria-pressed': isOpen,
+      className: classNames(
+        button.props.className,
+        'euiCollapsibleNav__toggle'
+      ),
+    });
+
+  const flyout = (
+    <>
       <EuiWindowEvent event="keydown" handler={onKeyDown} />
       {optionalOverlay}
-      {/* Trap focus only when isDocked={false} */}
-      <EuiFocusTrap disabled={isDocked} clickOutsideDisables={true}>
-        <nav className={classes} role="group" {...rest}>
+      {/* Trap focus only when docked={false} */}
+      <EuiFocusTrap disabled={navIsDocked} clickOutsideDisables={true}>
+        <nav id={flyoutID} className={classes} role="group" {...rest}>
```

Comment:
Sorry, must have missed this in a previous review, we shouldn't be overriding `nav` semantics

```suggestion
        <nav id={flyoutID} className={classes} {...rest}>
```

Reply:
Ah yeah I think that was an attempt to work out @snide's comment about group roles and forgot to remove.